modules/junit-jupiter/src/main/java/org/testcontainers/junit/jupiter/TestcontainersExtension.java

```diff
@@ -66,6 +65,22 @@ public void beforeAll(ExtensionContext context) {
         signalBeforeTestToContainers(lifecycleAwareContainers, testDescriptionFrom(context));
     }
 
+    private void startSharedContainers(
+        List<StoreAdapter> sharedContainersStoreAdapters,
+        Store store,
+        ExtensionContext context
+    ) {
+        if (isParallelExecutionEnabled(context)) {
+            Startables
+                .deepStart(sharedContainersStoreAdapters.stream().map(storeAdapter -> storeAdapter.container))
+                .join();
+        } else {
+            sharedContainersStoreAdapters.forEach(adapter -> {
+                store.getOrComputeIfAbsent(adapter.getKey(), k -> adapter.start());
+            });
```

Comment:
@eddumelendez updated the code here to utilize Startables and kept the previous logic as is